Source/WebKit/NetworkProcess/cache/NetworkCacheIOChannelCurl.cpp

```diff
@@ -57,7 +57,7 @@ IOChannel::~IOChannel()
 
 void IOChannel::read(size_t offset, size_t size, WTF::WorkQueueBase& queue, Function<void(Data&, int error)>&& completionHandler)
 {
-    queue.dispatch([this, protectedThis = Ref { *this }, offset, size, completionHandler = WTFMove(completionHandler)] {
+    queue.dispatchInternal([this, protectedThis = Ref { *this }, offset, size, completionHandler = WTFMove(completionHandler)] {
```

Comment:
These look very dubious. I think these could be tried to be fixed by using the correct type, e.g. WorkQueue?

Reply:
Unfortunately, not the way the method is used, it is passed both a ConcurrentWorkQueue and a WorkQueue. 
Only way to fix this issue would be to make either IOChannel::read/write a templated method, or (preferred as more elegant, logical) make WorkQueue inherit from ConcurrentWorkQueue.
I've put such implementation here https://github.com/WebKit/WebKit/compare/main...jyavenard:WebKit:WorkQueue
to be discussed further